### PR TITLE
Refactor/notulen publication

### PR DESCRIPTION
**Pull request by nvdk**
- [x] need to verify intermissions render ok in the notulen
- [x] need to fix the tests

so this turned into a large refactor, but it does fix some issues we had:
- if you sign and then publish previously content would contain the full notulen, even if treatments were marked private
- if you publish and then sign the reverse

both should now insert the correct content. this also removed a lot of duplicate code. it did become quite a monster of a PR.

### CODE REVIEW

File: routes/preview.js [hunk 1]
```diff
@@ -80,16 +82,63 @@ router.get('/prepublish/behandelingen/:zittingIdentifier', async function(req, r
...
+  catch(e) {
+    console.error(e);
+    const error = new Error(`An error occurred while fetching contents for prepublished notulen ${req.params.zittingIdentifier}: ${e}`);
     // @ts-ignore
     error.status = 500;
     return next(error);
   }
 });
 
+router.post('/extract-previews', async function (req, res, next) {
```
Comment (nvdk) [inline routes/preview.js:97]: this is a new json:api compliant route that can be used in the future. should work like this in ember:
```
const extractPreview = this.store.createRecord('extract-preview');
extractPreview.treatment = treatment;
yield extractPreview.save();
// extractPreview.html is now set

```

Comment (nvdk) [reply routes/preview.js:97]:
```
import Model from '@ember-data/model';
import { attr, belongsTo, hasMany } from '@ember-data/model';
export default class ExtractPreviewModel extends Model {
  @belongsTo('behandeling-van-agendapunt') treatment;
  @attr html;
}
```
tested with this model

**Comment by nvdk** [comment]
@lagartoverde thanks a lot for the tests on the notulen, I would not have been able to fully trust the refactor without them!

**Comment by nvdk** [comment]
seems I've lost the translations in the intermission

**Comment by lagartoverde** [comment]
I think it's my fault because I forgot to include them in the tests, but we are missing the participation list in the meeting, aren't we?

**Review by abeforgit — COMMENTED** [review]
You mention this fixes:
> if you sign and then publish previously content would contain the full notulen, even if treatments were marked private

but it seems to be worse? in that any content marked private is still included in the "publication preview" modal
this could be a GN issue maybe

**Review by abeforgit — APPROVED** [review]
Code looks good on first glance, to big to dig in, but I couldn't find any regressions

**Comment by nvdk** [comment]
> You mention this fixes:
> 
> > if you sign and then publish previously content would contain the full notulen, even if treatments were marked private
> 
> but it seems to be worse? in that any content marked private is still included in the "publication preview" modal
> this could be a GN issue maybe
> 
> edit: it's not worse, this is the same on dev

Yeah this will require quite a bit of work in GN to fix, but I think in the backend at least the abstractions are in place to easily provide a new api endpoint to request the correct view

**Pull request merged**